### PR TITLE
Update key generation

### PR DESCRIPTION
**Pull request by KomelT**
#272

### CODE REVIEW

**Comment by Hunter275** [comment]
@KomelT 

We shouldn't need to generate the Admin Key, it will be the key of another device.

We should also add a confirmation dialog for generating a new Private Key.

I will add these to #272

**Comment by KomelT** [comment]
@Hunter275 yes you are right, I miss-read the comment. Thanks.

File: src/core/utils/x25519.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { x25519 } from "@noble/curves/ed25519";
+
+export function getX25519PrivateKey(): Uint8Array {
+  return x25519.utils.randomPrivateKey();
```
Comment (jp-bennett) [inline src/core/utils/x25519.ts]: I suspect this needs the bit manipulation of the top and bottom bits: 
       k[0] &= 248
       k[31] &= 127
       k[31] |= 64

Comment (Hunter275) [reply src/core/utils/x25519.ts]: @KomelT if this is working now please resolve this comment

**Comment by jp-bennett** [comment]
Key generation seems to be working as expected now. There are a couple little issues I've seen. First, having legacy admin enabled actually disables the private key field, where it should be disabling the admin key. (Though technically both can be turned on if somebody really wanted to).

Next, I don't understand the submit button, as none of the other fields have this. It does seem to be required to get settings to stick, to hit submit first, then the save icon. 

~~And finally, if a user pasts a private key in to the field, this should blank out the public key, so the firmware will regenerate the appropriate key.~~ Nevermind, the web app correctly regenerates the public key, and I didn't notice. 0=)

File: src/components/PageComponents/Config/Security.tsx [hunk 2]
```diff
@@ -50,191 +51,200 @@ export const Security = (): JSX.Element => {
...
+                bits: [{ text: "256 bit", value: "32", key: "bit256" }],
+                validationText: privateKeyValidationText,
+                devicePSKBitCount: privateKeyBitCount,
+                inputChange: privateKeyInputChangeEvent,
+                selectChange: privateKeySelectChangeEvent,
+                hide: !privateKeyVisible,
+                buttonClick: privateKeyClickEvent,
+                disabledBy: [
+                  {
+                    fieldName: "adminChannelEnabled",
```
Comment (Hunter275) [inline src/components/PageComponents/Config/Security.tsx]: This should only be on the Admin key, not the Private key

**Review by Hunter275 — APPROVED** [review]

**Pull request merged**